File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        
        if n < 3:
            return n
        
        first = 1
        second = 2
        
        for i in range(2, n):
            temp = first+second
            first = second
            second = temp

        return second
        
        
        
#         # another solution
#         array = [1, 2, 3]
#         for i in range(3, n):
#             array.append(array[i-1] + array[i-2])
#         return array[len(array)-1]

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
        if (n <= 3) {
            return n;
        }
        
        int first = 2;
        int second = 3;
        int result = 0;
        
        for(int i = 4; i <= n; i++) {
            result = first + second;
            first = second;
            second = result;
        }
        
        return result;
        
        
        
        
        
        
        
        // using DP approach
//         int* arr= new int[n+1];
        
//         arr[1] = 1;
//         arr[2] = 2;
//         arr[3] = 3;
        
//         for(int i = 4; i <= n; i++) {
//             arr[i] = arr[i-1] + arr[i-2];
//         }
        
//         int result = arr[n];
        
//         delete[] arr;
        
//         return result;
        
    }
};